### PR TITLE
feat(prow/github): add client.GetPullRequestDiff method

### PR DESCRIPTION
**Pull request by wuhuizuo**
- add `client.GetPullRequestDiff`.
- update http header and update comments.

### CODE REVIEW

File: prow/github/client.go [hunk 1]
```diff
@@ -2298,15 +2298,32 @@ func (c *client) EditIssue(org, repo string, number int, issue *Issue) (*Issue,
...
 }
 
+// GetPullRequestDiff gets the diff version of a pull request.
+//
+// See https://docs.github.com/zh/rest/overview/media-types?apiVersion=2022-11-28#commits-commit-comparison-and-pull-requests
+func (c *client) GetPullRequestDiff(org, repo string, number int) ([]byte, error) {
+	durationLogger := c.log("GetPullRequestDiff", org, repo, number)
+	defer durationLogger()
+
+	_, patch, err := c.requestRaw(&request{
```
Comment (smg247) [inline prow/github/client.go:2308]: rename `patch` to `diff`

File: prow/github/client.go [hunk 2]
```diff
@@ -2298,15 +2298,32 @@ func (c *client) EditIssue(org, repo string, number int, issue *Issue) (*Issue,
 	return &ret, nil
 }
 
+// GetPullRequestDiff gets the diff version of a pull request.
+//
+// See https://docs.github.com/zh/rest/overview/media-types?apiVersion=2022-11-28#commits-commit-comparison-and-pull-requests
+func (c *client) GetPullRequestDiff(org, repo string, number int) ([]byte, error) {
```
Comment (smg247) [inline prow/github/client.go:2304]: Be sure to add this method to the `PullRequestClient` interface

**Comment by wuhuizuo** [comment]
> /ok-to-test

@smg247 Thank you! I have solved them.

**Review by matthyx — APPROVED** [review]
/lgtm
At some point it would be good to have some test coverage for these kinds of methods...

File: prow/github/client.go [hunk 3]
```diff
@@ -2298,15 +2299,32 @@ func (c *client) EditIssue(org, repo string, number int, issue *Issue) (*Issue,
...
+		org:       org,
+		exitCodes: []int{200},
+	})
+	return diff, err
+}
+
 // GetPullRequestPatch gets the patch version of a pull request.
 //
-// See https://developer.github.com/v3/media/#commits-commit-comparison-and-pull-requests
+// See https://docs.github.com/zh/rest/overview/media-types?apiVersion=2022-11-28#commits-commit-comparison-and-pull-requests
```
Comment (petr-muller) [inline prow/github/client.go:2321]: Please make both these links point to an english version of the docs, not the chinese one

**Review by petr-muller — APPROVED** [review]

**Comment by petr-muller** [comment]
Thanks!

**Pull request merged**